rosbag2_transport/src/rosbag2_transport/generic_publisher.cpp

```diff
@@ -48,4 +49,34 @@ void GenericPublisher::publish(std::shared_ptr<rmw_serialized_message_t> message
   }
 }
 
+void GenericPublisher::publish_loaned_message(void * ros_message)
```

Comment:
I am having a bit of a problem with this signature. First off, I don't really like to publish a void pointer. That gives us really the least amount of type safety here. The least we should do is to verify that it's not null.
But then further, this function actually requires a full-fleshed ROS message, opposing to a serialized message like the rest of the publishing functions. What's the benefit here?

I would actually see that there should be a `SerializedLoanedMessage` and the appropriate set of RMW functions to loan a naive data buffer. While with this approach you might safe a copy in the initialization phase, you add another two complex steps (deserialize message to ROS2, copy ros2 into loaned message).

Reply:
@Karsten1987 

I agree that a typed interface would be better.  I am just using the interface defined in rcl (https://github.com/ros2/rcl/blob/master/rcl/include/rcl/publisher.h#L407) though. This interface takes a `void * ros_message`. The `ros_message` is not dereferenced in `GenericPublisher::publish_loaned_message`, that's why I did not check for a nullptr. `rcl_publish_loaned_message` does check for nullptr.

What would you suggest here?